src/botMessage/dialogFlow.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@nestjs/common';
+const dialogflow = require('dialogflow'); // eslint-disable-line
+
+@Injectable()
+export class DialogFlowService {
+  async getDialogflowIntent(
+    conversationId: string,
+    userMessage: string,
+  ): Promise<string> {
+    let privateKey;
+    if (process.env.NODE_ENV === 'production'){
+      privateKey = JSON.parse(process.env.DIALOGFLOW_PRIVATE_KEY)
+    } else if (process.env.NODE_ENV === 'travis') {
+      privateKey = process.env.DIALOGFLOW_PRIVATE_KEY.replace(/\\n/gm, '\n')
+    } else {
+      process.env.DIALOGFLOW_PRIVATE_KEY
+    }
+
+    const clientEmail = process.env.DIALOGFLOW_CLIENT_EMAIL;
+    const config = {
+      credentials: {
+        'private_key': privateKey,
+        'client_email': clientEmail,
+      },
+    };
+    const sessionClient = new dialogflow.SessionsClient(config);
+    const sessionPath = sessionClient.sessionPath(
+      process.env.DIALOGFLOW_PROJECT_ID,
+      conversationId,
+    );
+
+    // TODO: set language Code
```

Comment:
Looks like this `TODO` could be removed now?

Reply:
Not yet! I need to do some French updates then we'll have to switch dialogflow back and forth.

Reply:
Ah! Good to know. The comment could do with a little more elaboration (based on what you said). I know I would forgot in a few weeks times if/when I see this comment 😄